client/python/openlineage/client/serde.py

```diff
@@ -33,7 +33,7 @@ def remove_nulls_and_enums(cls, obj: Any) -> Any:  # noqa: ANN401
         if isinstance(obj, List):
             return list(
                 filter(
-                    lambda x: x is not None and (isinstance(x, dict) and x != {}),
+                    lambda x: False if x is None else not (isinstance(x, dict) and x == {}),
```

Comment:
I really don't like returning booleans with ternary operator in Python.
```suggestion
                    lambda x: x is not None and (not isinstance(x, dict) or x != {}),
```